internal/events/event.go

```diff
@@ -14,16 +14,25 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-//go:generate mockgen -source=event.go -package=events -destination=mock_event.go
-
-type Handler interface {
-	//Add event record to the event table. Use the prop field to add list of arbitrary key value pairs
-	//when additional information is needed (for example: "vendor": "RedHat")
+type Sender interface {
+	// AddEvent add an event for and entityID.
+	// Since events, might relate to multiple entities, for example:
+	//     host added to cluster, we have the host-ID as the main entityID and
+	//     the cluster-ID as another ID that this event should be related to
+	// Use the prop field to add list of arbitrary key value pairs when additional information is needed (for example: "vendor": "RedHat")
 	AddEvent(ctx context.Context, clusterID strfmt.UUID, hostID *strfmt.UUID, severity string, msg string, eventTime time.Time, props ...interface{})
+
 	//Add metric-related event. These events are hidden from the user and has 'metrics' Category field
 	AddMetricsEvent(ctx context.Context, clusterID strfmt.UUID, hostID *strfmt.UUID, severity string, msg string, eventTime time.Time, props ...interface{})
-	//Get a list of events. Events can be filtered by category. if no filter is specified,
-	//events with the default category are returned
+
+	SendClusterEvent(ctx context.Context, event ClusterEvent, eventTime time.Time)
```

Comment:
It seems that > 95% of the time we call this method with time.Now().
What do you think about adding a method that doesn't expect eventTime and fires it immediately ? 
e.g 
```
SendClusterEvent(ctx context.Context, event ClusterEvent)
SendClusterEventAtTime(ctx context.Context, event ClusterEvent, eventTime time.Time)
```
same for host-events

Reply:
sounds good.
in the test files it's always gomock.Any(). should I create also sendClusterEventAtAnyTime() for the tests?
(same for host)

Reply:
solved.
the default time for the mock events is gomock.Any().
also, I think time.Now() is the time for the real events and gomock.Any() is the time for the mock events in 100% of the cases so I'm not sure why we need to keep the option of time as an input